File: DataInjectorIOS/DataInjector/Classes/dependency/InjectorDependencyState.swift
//
//  InjectorDependencyState.swift
//  DataInjector Pod
//
//  Library dependency utility: the state
//  Used to keep track of the dependency state
//

import Foundation

/// The state of the dependency using values which can be used as flags for bitwise comparisons
public enum InjectorDependencyState: Int {
    
    case idle = 0
    case loading = 1
    case resolved = 2
    case refreshing = 3
    case error = 4
    case refreshError = 6
    
    func isKindOfState(_ state: InjectorDependencyState) -> Bool {
        if state == .idle && self == .idle {
            return true
        }
        return (rawValue & state.rawValue) == state.rawValue
    }

}

File: DataInjectorAndroid/DataInjectorLib/src/main/java/com/crescentflare/datainjector/dependency/InjectorDependencyState.java
package com.crescentflare.datainjector.dependency;

/**
 * Data injector dependency: the state
 * Used to keep track of the dependency state
 */
public enum InjectorDependencyState
{
    Idle(0),
    Loading(1),
    Resolved(2),
    Refreshing(3),
    Error(4),
    RefreshError(6);

    private int intValue;

    InjectorDependencyState(int intValue)
    {
        this.intValue = intValue;
    }

    public boolean isTypeOfState(InjectorDependencyState state)
    {
        if (state == Idle && this == Idle)
        {
            return true;
        }
        return (intValue & state.intValue) == state.intValue;
    }
}
